src/vdaf.rs

```diff
@@ -248,6 +255,38 @@ pub trait Aggregator<const VERIFY_KEY_SIZE: usize, const NONCE_SIZE: usize>: Vda
         agg_param: &Self::AggregationParam,
         output_shares: M,
     ) -> Result<Self::AggregateShare, VdafError>;
+
+    #[cfg(feature = "experimental")]
+    /// Adds noise to an aggregate share to achieve differential privacy.
+    /// The noise is drawn from the `dist` distribution, which contains
+    /// the relevant DP parameters.
+    ///
+    /// TODO(tholop): VDAFs that don't want DP can make it a no-op.
+    ///               How about VDAFs that sometimes want DP and sometimes not?
+    ///               Leave it up to them how they implement their `dist` parameter,
+    ///               potentially with a `None` variant?
```

Comment:
Something we need to avoid is silently violating the user's expectations: If they call this method, then they expect the aggregate result to be DP. If an `Aggregator` doesn't actually implement DP (yet), then either the code should fail to compile or there should be an error at runtime.

Another option: Add yet another trait:

```rust
pub trait NoiseAdder: Aggregator {
    fn add_noise_to_agg_share();
}
```

What do folks think of this?

Reply:
Good question, but to refine it a bit more: should *server-DP* be optional, since this is what we're implementing here.

I think that it needs to be optional. There are a variety of DP mechanisms that may be compatible with a given VDAF, and I don't think we can or should try to encourage one over another. Ultimately the choice of DP mechanism comes down to the application and is not specific to the VDAF.

Reply:
(@divergentdave see this thread as well)